chat/src/main/java/io/skygear/plugins/chat/CacheController.java

```diff
@@ -0,0 +1,79 @@
+package io.skygear.plugins.chat;
+
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Date;
+import java.util.List;
+
+import io.realm.RealmQuery;
+
+class CacheController {
+    private static CacheController sharedInstance;
+
+    static CacheController getInstance() {
+        if (sharedInstance == null) {
+            RealmStore store = new RealmStore("SKYChatCache", false);
+            sharedInstance = new CacheController(store);
+        }
+
+        return sharedInstance;
+    }
+
+    RealmStore store;
+
+    CacheController(RealmStore store) {
+        this.store = store;
+    }
+
+    void getMessages(@NonNull final Conversation conversation,
+                            final int limit,
+                            @Nullable final Date before,
+                            @Nullable final String order,
+                            @Nullable final GetCallback<List<Message>> callback) {
+        RealmStore.QueryBuilder<MessageCacheObject> queryBuilder = new RealmStore.QueryBuilder<MessageCacheObject>() {
+            @Override
+            public RealmQuery<MessageCacheObject> buildQueryFrom(RealmQuery<MessageCacheObject> baseQuery) {
+                RealmQuery<MessageCacheObject>  query = baseQuery
```

Comment:
@rickmak need your help review the logic for this query

@Steven-Chan deleted the extra space between `>` and `query`